src/steps/os/macos.rs

```diff
@@ -93,3 +94,79 @@ pub fn run_sparkle(ctx: &ExecutionContext) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn update_xcodes(ctx: &ExecutionContext) -> Result<()> {
+    let xcodes = require("xcodes")?;
+    print_separator("Xcodes");
+
+    let should_ask = !(ctx.config().yes(Step::System)) || (ctx.config().dry_run());
```

Comment:
Seems that it should be the `Xcodes` step, and the condition should be something like:

```suggestion
    let should_ask = !(ctx.config().yes(Step::Xcodes) || ctx.config().dry_run());
```

Reply:
It seems that you referred to the implementation of the System step on macOS while implementing this feature, and given that you are a macOS user, would you like to take a look at #543, [changing the definition of `should_ask` should fix it](https://github.com/topgrade-rs/topgrade/issues/543#issuecomment-1897676440), I would like to file a PR for it but I don't have a macOS machine at hand so I cannot give it a test

Reply:
Can do, should I open a separate PR or resolve the issue here?

Reply:
Yep, it should be done in another PR, thanks for your help:)

Reply:
Seems that the condition is till not correct? `should_ask` should be false if either `ctx.config().yes(Step::Xcodes)` or `ctx.config().dry_run()` is `true`

Reply:
Exactly, I'll implement the change.

Reply:
#668 :^)